c7n/resources/sns.py

```diff
@@ -252,3 +255,69 @@ def process(self, resources):
                 AttributeValue=json.dumps(policy)
             )
         return results
+
+
+@SNS.filter_registry.register('kms-key')
+class KmsFilter(KmsRelatedFilter):
+
+    RelatedIdsExpression = 'KmsMasterKeyId'
+
+
+@SNS.action_registry.register('set-encryption')
+class SetEncryption(BaseAction):
+    """
+    Set Encryption on SNS Topics
+
+    By default if no key is specified, alias/aws/sns is used
+
+    key can either be a KMS key ARN or an alias, when using aliases
+    prefix the key with 'alias/', i.e.: 'alias/custom/key'.
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: set-sns-topic-encryption
+              resource: sns
+              actions:
+                - type: set-encryption
+                  key: alias/cmk/key
+                  enabled: True
+
+    """
+    schema = type_schema(
+        'set-encryption',
+        enabled={'type': 'boolean'},
+        key={'type': 'string'}
+    )
+
+    permissions = ('sns:SetTopicAttributes', 'kms:DescribeKey',)
+
+    def process(self, resources):
+        sns = local_session(self.manager.session_factory).client('sns')
+
+        if self.data.get('enabled', True):
+            key = self.data.get('key', 'alias/aws/sns')
+            if key != 'alias/aws/sns':
```

Comment:
so the service accepts 'alias/aws/sns' but needs an Arn for any other key?

Reply:
actually no, I was testing with the incorrect alias which resulted in the key not being found. After some more testing it looks like you can supply alias/key, the key id, or the arn of the key. I'll be updating it to remove this as it's incorrect.